dotnet/src/Agents/OpenAI/Internal/AssistantThreadActions.cs

```diff
@@ -324,66 +328,144 @@ async Task PollRunStatusAsync()
 
             logger.LogOpenAIAssistantPolledRunStatus(nameof(PollRunStatusAsync), run.Status, run.Id, threadId);
         }
+    }
 
-        // Local function to capture kernel function state for further processing (participates in method closure).
-        IEnumerable<FunctionCallContent> ParseFunctionStep(OpenAIAssistantAgent agent, RunStep step)
+    /// <summary>
+    /// Invoke the assistant on the specified thread using streaming.
+    /// </summary>
+    /// <param name="agent">The assistant agent to interact with the thread.</param>
+    /// <param name="client">The assistant client</param>
+    /// <param name="threadId">The thread identifier</param>
+    /// <param name="messages">The receiver for the completed messages generated</param>
+    /// <param name="invocationOptions">Options to utilize for the invocation</param>
+    /// <param name="logger">The logger to utilize (might be agent or channel scoped)</param>
+    /// <param name="kernel">The <see cref="Kernel"/> plugins and other state.</param>
+    /// <param name="arguments">Optional arguments to pass to the agents's invocation, including any <see cref="PromptExecutionSettings"/>.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Asynchronous enumeration of messages.</returns>
+    /// <remarks>
+    /// The `arguments` parameter is not currently used by the agent, but is provided for future extensibility.
+    /// </remarks>
+    public static async IAsyncEnumerable<StreamingChatMessageContent> InvokeStreamingAsync(
+        OpenAIAssistantAgent agent,
+        AssistantClient client,
+        string threadId,
+        IList<ChatMessageContent> messages,
+        OpenAIAssistantInvocationOptions? invocationOptions,
+        ILogger logger,
+        Kernel kernel,
+        KernelArguments? arguments,
```

Comment:
Taking into account the number of parameters and also `arguments` parameter that is added for future extensibility only, should we introduce a separate class that will contain all of these parameters? `cancellationToken` parameter should still be here, but other parameters can live in separate class, in this case we don't need to add `arguments` parameter now, but we will always be able to add it later without breaking changes.

Reply:
I do entirely agree on the class encapsulation obsevation...generally method with 100 parameters is missing an encapuslation.  I have created a bunch of settings encapsulation for just this very reason.

In this case, this method is internal only and invoked from only two call sites for with a good portion of these parameters are aligned with the current signatures.  At this point, the method should be fairly stable, but if it does change in the future...it won't be breaking for any user of the framework as it is internal.  

My may thoughts on wrapping everything in an internal-only object just to pass to this method it that would require introducing a new internal calss / more code to maintain...for marginal impact.  Curious on your thoughts.

Reply:
In my opinion, if it's internal, in this case we don't need to accept `arguments` parameter now, as it's not used, and we can always add it later without breaking changes.

Reply:
I am currently working on a feature that uses this.  Matthew was keen (insisted) on plumbing the "arguments" parameter through everything in advance.  (it is already present on the non-streaming invocation method)